### ISSUE TITLE
Running btrbk with -q produces "At subvol..." output

### ISSUE THREAD

**Issue by timharton** [digint/btrbk#33]
Running btrbk 0.19.3.

Running btrbk via cron sends me an email even if there aren't any errors.
I tried running the command outside of cron gives me:
 
```
sudo btrbk -q -c /etc/btrbk/btrbk-######.conf run           
At subvol /mnt/btr_system/btrbk_snapshots/@.20150709_3
At subvol /mnt/btr_system/btrbk_snapshots/@home.20150709_3
At subvol /mnt/btr_system/btrbk_snapshots/@vms.20150709_3
```

Which is the same output I'm being emailed daily.
I'm trying to run btrfs-tools only from the main repo (3.17) on Kubuntu 15.04 on my desktop. This doesn't happen on my main ubuntu server where I run the latest btrfs-progs.
I guess it might not be a btrfs-tools or btrbk bug and just a problem with my system.
If this isn't an easy fix I can just download and build the latest btrfs-progs, but I thought I should let you know.

**Comment by timharton** [comment]
Re ran btrbk with these settings:
```
sudo /usr/sbin/btrbk -q -v -l info -c /etc/btrbk/btrbk-#####.conf run 
```

This is the first part of the output:

```
btrbk command line client, version 0.19.3  (Thu Jul  9 11:04:04 2015)
Using configuration: /etc/btrbk/btrbk-#######.conf
Creating subvolume snapshot for: /mnt/btr_system/@
>>> /mnt/btr_system/btrbk_snapshots/@.20150709_4
Creating subvolume snapshot for: /mnt/btr_system/@home
>>> /mnt/btr_system/btrbk_snapshots/@home.20150709_4
Creating subvolume snapshot for: /mnt/btr_system/@vms
>>> /mnt/btr_system/btrbk_snapshots/@vms.20150709_4
Checking for missing backups of subvolume "/mnt/btr_system/@" in: {#######}/mnt/btrfs/btrbk/#####/
No missing backups found
Creating subvolume backup (send-receive) for: /mnt/btr_system/@
Receiving from snapshot: /mnt/btr_system/btrbk_snapshots/@.20150709_4
Incremental from parent snapshot: /mnt/btr_system/btrbk_snapshots/@.20150709_3
>>> {#######}/mnt/btrfs/btrbk/#####/@.20150709_4
At subvol /mnt/btr_system/btrbk_snapshots/@.20150709_4
```
It seems to be occurring after sending an incremental snapshot to my server.

**Comment by digint** [comment]
Even with the latest btrfs-progs, you will still see the `At subvol ...` messages. These are generated by `btrfs send` on stderr, which btrbk does not catch and process. This is an issue and I won't fix this on btrbk, since we want to see the real errors on stderr if something fails. I will try to provide patches to btrfs-progs for disabling this, but for now we all have to live with it...

The `-q` option only disables the summary display at the end of processing, and does not affect stderr output (btrbk messages as well as output from `btrfs` commands).

**Comment by digint** [comment]
I fixed this logging problem by catching and filtering stderr from external commands (btrfs-progs). It's on the ["open3" branch](https://github.com/digint/btrbk/tree/open3), and I will merge this to master as soon as I performed some more testing on several error cases.

I would apreciate if you (and also my GitHub followers reading this) could give it a try, I'm especially interested on how this behaves on old Perl versions (I'm using perl-5.20.2, which comes with IPC::Open3 v1.16 included on my system).

**Comment by timharton** [comment]
Will do

**Comment by digint** [comment]
After digging deeper into the perl and file descriptor hell, I rewrote the stderr catching code, which is now in the ["run_cmd_open3" branch](https://github.com/digint/btrbk/tree/run_cmd_open3) (the old "open3" branch got deleted).
I did much testing, and it still produces deadlocks in some special test cases. I could not observe deadlocks with normal btrbk calls though, but as long as I'm not convinced that everything is correct I will not merge this.
